app/main.py

```diff
@@ -1,3 +1,23 @@
-def shop_trip():
-    # write your code here
-    pass
+from app.customer import Customer
+from app.load_jsons import load_complete_json
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    data = load_complete_json("app/config.json")
+    shops = [Shop(**shop) for shop in data.get("shops")]
+    fuel_price = data.get("FUEL_PRICE")
+
+    for customer_data in data.get("customers"):
+        customer = Customer(**customer_data)
+        best_price, best_shop = customer.choose_the_cheapest_shop_visit(
```

Comment:
here you could move all those methods calls on customer to a separate method (e.g. complete_shop_trip ) and call these methods in a chain inside of the class. Here you would just call complete_shop_trip on Customer and thats it. 
This way you will cleanup this function